string u
string v
int8 state
float32 distance
